File: 0225-implement-stack-using-queues/0225-implement-stack-using-queues.py
from Queue import Queue

class MyStack(object):

    def __init__(self):
        self.q1 = Queue()
        self.q2 = Queue()

    def push(self, x):
        """
        :type x: int
        :rtype: None
        """
        s = self.q1.qsize()
        if self.q1.empty():
            self.q1.put(x)
        
        elif not self.q1.empty() and self.q2.empty():
            while not self.q1.empty():
                temp = self.q1.get()
                self.q2.put(temp)
            
            self.q1.put(x)

            while not self.q2.empty():
                temp = self.q2.get()
                self.q1.put(temp)
        
        

    def pop(self):
        """
        :rtype: int
        """
        return self.q1.get()

    def top(self):
        """
        :rtype: int
        """
        return self.q1.queue[0]

    def empty(self):
        """
        :rtype: bool
        """
        if self.q1.empty():
            return True
        else:
            return False
        


# Your MyStack object will be instantiated and called as such:
# obj = MyStack()
# obj.push(x)
# param_2 = obj.pop()
# param_3 = obj.top()
# param_4 = obj.empty()

File: 0225-implement-stack-using-queues/0225-implement-stack-using-queues.java
import java.util.LinkedList;
import java.util.Queue;
class MyStack {
        Queue<Integer> q1;
        Queue<Integer> q2;
    public MyStack() {
        q1 = new LinkedList<>();
        q2 = new LinkedList<>();        
    }
    
    public void push(int x) {
        int size = q1.size();
        if(q1.isEmpty()){
            q1.offer(x);
        }
        else if(!q1.isEmpty() && q2.isEmpty()){
            for(int i=1; i<=size; i++){
                int temp = q1.poll();
                q2.offer(temp);
            }
            q1.offer(x);
            while(!q2.isEmpty()){
                int temp = q2.poll();
                q1.offer(temp);
            }
        }
    }
    
    public int pop() {
        return q1.poll();
    }
    
    public int top() {
        return q1.peek();
    }
    
    public boolean empty() {
        if(q1.isEmpty()){
            return true;
        }
        else{
            return false;
        }
    }
}

/**
 * Your MyStack object will be instantiated and called as such:
 * MyStack obj = new MyStack();
 * obj.push(x);
 * int param_2 = obj.pop();
 * int param_3 = obj.top();
 * boolean param_4 = obj.empty();
 */